### PR TITLE
[Excel] (Custom functions) Clears up difference between stream and cancelable

### PR DESCRIPTION
**Pull request by mscharlock**
Changes wrt to customer confusion about the difference (in-mail last week). Hopefully this makes the difference between the two clearer. Are there other areas where we should highlight the difference besides here that you can think of?

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[15940cb](https://github.com/OfficeDev/office-js-docs-pr/pull/1037/commits/15940cb768df599ddd595a3736c64e0dfe9f4615)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-web-reqs.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-web-reqs.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C5%5C24%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201905241956440110-1037%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=EhiBKORVQfc9dDtGgWZ6Xo3arbOuGnywRquvDOFRD3Q%3D&st=2019-05-24T19%3A53%3A02Z&se=2019-06-24T19%3A58%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/custom-functions-web-reqs.md [hunk 1]
```diff
@@ -134,11 +134,9 @@ ws.onerror(error){
 
 ## Stream and cancel functions
 
-Streaming custom functions enable you to output data to cells that updates repeatedly, without requiring a user to explicitly refresh anything.
+Streaming custom functions enable you to output data to cells that updates repeatedly, without requiring a user to explicitly refresh anything. Streaming might be online or offline, such as a function which counts a number up by one each second (offline) or a function which requests data from the web at set intervals (online). To declare a streaming function, use the JSDoc comment tag `@stream`.
```
Comment (davidchesnut) [inline docs/excel/custom-functions-web-reqs.md]: Custom functions will not work when Excel is offline. I think this is not really necessary to specify. Any custom function can call the web, and they can be streaming, or not streaming. For example you could write all custom functions to call Azure functions that have the real calculation code. But they might all return a single value (not streaming). Also the header is odd. There is no such thing as a "cancel function" but the header sort of sounds like these are two different things.

Maybe for the H1 use "Make your streaming function cancelable" or similar.

File: docs/excel/custom-functions-web-reqs.md [hunk 2]
```diff
@@ -134,11 +134,9 @@ ws.onerror(error){
...
 
-Cancelable custom functions enable you to cancel the execution of a streaming custom function to reduce its bandwidth consumption, working memory, and CPU load.
-
-To declare a function as streaming or cancelable, use the JSDOC comment tags `@stream` or `@cancelable`.
+Cancelable custom functions enable you to cancel the execution of an asynchronous custom function to reduce its bandwidth consumption, working memory, and CPU load. Declare them using the tag `@cancelable`.  Because cancelable functions are only asynchronous functions which are *not* streaming, you will likely use this tag much less frequently. An example usage of a cancelable function would be a function takes a single input and returns a single value based on a web request. This operation isn't streaming (because it returns a single value), but you need to make the function cancelable to address a situation where the single input changes before the web request can be completed. Streaming functions handle this situation by default and therefore do not need to be tagged as both streaming and cancelable.
```
Comment (davidchesnut) [inline docs/excel/custom-functions-web-reqs.md]: As the developer, you can't choose to cancel your function (first sentence could be read that way.) Excel will cancel based on a user action (like deleting the cell where your custom function was operating.) So in those sorts of cases you want to handle the cancel call from Excel or your code will just keep running even though there is no cell anymore listening to it. I would say maybe rewrite in context of an actual example (user deleted your cell) as to why you should respond to the cancel call from Excel (you don't want to continue running and sending data that Excel just ignores). Again, I would not refer to "cancelable custom functions" as though they are a special or separate thing. It's just something you want to enable on a streaming function.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[99b4417](https://github.com/OfficeDev/office-js-docs-pr/pull/1037/commits/99b44179125984e72508a19ccab0b93c75eb659f)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-parameter-options.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-parameter-options.md) | :warning:Warning | | [Details](#user-content-31e38006-c4d49b91)
[docs/excel/custom-functions-json.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-json.md) | :warning:Warning | | [Details](#user-content-6a119d07-c4d49b91)
[docs/excel/custom-functions-web-reqs.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-web-reqs.md) | :white_check_mark:Succeeded | |

<a id="31e38006-c4d49b91"></a>
### [docs/excel/custom-functions-parameter-options.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-parameter-options.md)
 - **Line 96**: **[Warning]** `````Illegal link: `<a href="excel/custom-functions-web-reqs.md#stream-and-cancel-functions">canceling a function</a>` -- missing bookmark. The file excel/custom-functions-web-reqs.md doesn't contain a bookmark named stream-and-cancel-functions.`````

<a id="6a119d07-c4d49b91"></a>
### [docs/excel/custom-functions-json.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-json.md)
 - **Line 133**: **[Warning]** `````Illegal link: `<a href="excel/custom-functions-web-reqs.md#stream-and-cancel-functions">Canceling a function</a>` -- missing bookmark. The file excel/custom-functions-web-reqs.md doesn't contain a bookmark named stream-and-cancel-functions.`````
 - **Line 135**: **[Warning]** `````Illegal link: `<a href="excel/custom-functions-web-reqs.md#stream-and-cancel-functions">Streaming functions</a>` -- missing bookmark. The file excel/custom-functions-web-reqs.md doesn't contain a bookmark named stream-and-cancel-functions.`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C5%5C30%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201905302223253347-1037%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=6wSaT%2Fq86UP69thjG%2F60PeMVRtZz68r9cZ1qEhpnVNY%3D&st=2019-05-30T22%3A19%3A30Z&se=2019-06-30T22%3A24%3A30Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[18c455c](https://github.com/OfficeDev/office-js-docs-pr/pull/1037/commits/18c455ceb0f6e5594a0cee5e426c86bee60feb5b)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-json.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-json.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-parameter-options.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-parameter-options.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-web-reqs.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-fixstreaming/docs/excel/custom-functions-web-reqs.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C5%5C30%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201905302246462763-1037%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=oCo%2BKsndNpZs17HvL2gWVGE7Sd5RgEUMezsvsB4HI0A%3D&st=2019-05-30T22%3A42%3A48Z&se=2019-06-30T22%3A47%3A48Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**